feat(route): add CQU jwc notification

AhsokaTano26 opened the pull request:
<!-- 
If you have any difficulties in filling out this form, please refer to https://docs.rsshub.app/joinus/new-rss/submit-route
如果你在填写此表单时遇到任何困难，请参考 https://docs.rsshub.app/zh/joinus/new-rss/submit-route
-->

## Involved Issue / 该 PR 相关 Issue

Close #

## Example for the Proposed Route(s) / 路由地址示例

<!--
Please include route starts with /, with all required and optional parameters.
Fail to comply will result in your pull request being closed automatically.
请在 `routes` 区域填写以 / 开头的完整路由地址，否则你的 PR 将会被无条件关闭。
如果路由包含在文档中列出可以完全穷举的参数（例如分类），请依次全部列出。

```routes
/some/route
/some/other/route
/dont/use/this/or/modify/it
/use/the/fenced/code/block/below
```

If your changes are not related to route, please fill in `routes` section with `NOROUTE`. Fail to comply will result in your PR being closed.
如果你的 PR 与路由无关, 请在 `routes` 区域 填写 `NOROUTE`，而不是直接删除 `routes` 区域。否则你的 PR 将会被无条件关闭。
-->

```routes
/cqu/jwc
```

## New RSS Route Checklist / 新 RSS 路由检查表
  
- [x] New Route / 新的路由
  - [x] Follows [Script Standard](https://docs.rsshub.app/joinus/advanced/script-standard) / 跟随 [路由规范](https://docs.rsshub.app/zh/joinus/advanced/script-standard)
- [ ] Anti-bot or rate limit / 反爬/频率限制
  - [ ] If yes, do your code reflect this sign? / 如果有, 是否有对应的措施?
- [x] [Date and time](https://docs.rsshub.app/joinus/advanced/pub-date) / [日期和时间](https://docs.rsshub.app/zh/joinus/advanced/pub-date)
  - [x] Parsed / 可以解析
  - [x] Correct time zone / 时区正确
- [ ] New package added / 添加了新的包
- [ ] `Puppeteer`

## Note / 说明
本路由提供对于重庆大学本科教学信息网通知的获取

AhsokaTano26 committed:
## Involved Issue / 该 PR 相关 Issue

Close #

## Example for the Proposed Route(s) / 路由地址示例
```routes
/cqu/jwc
```

## New RSS Route Checklist / 新 RSS 路由检查表

- [x] New Route / 新的路由
- [x] Follows [Script Standard](https://docs.rsshub.app/joinus/advanced/script-standard) / 跟随 [路由规范](https://docs.rsshub.app/zh/joinus/advanced/script-standard)
- [ ] Documentation / 文档说明
- [x] Full text / 全文获取
- [ ] Use cache / 使用缓存
- [ ] Anti-bot or rate limit / 反爬/频率限制
- [ ] If yes, do your code reflect this sign? / 如果有, 是否有对应的措施?
- [x] [Date and time](https://docs.rsshub.app/joinus/advanced/pub-date) / [日期和时间](https://docs.rsshub.app/zh/joinus/advanced/pub-date)
- [x] Parsed / 可以解析
- [x] Correct time zone / 时区正确
- [ ] New package added / 添加了新的包
- [ ] `Puppeteer`

## Note / 说明

AhsokaTano26 committed:
feat(route):add CQU jwc notification
## Involved Issue / 该 PR 相关 Issue

Close #

## Example for the Proposed Route(s) / 路由地址示例
```routes
/cqu/jwc
```

## New RSS Route Checklist / 新 RSS 路由检查表

- [x] New Route / 新的路由
- [x] Follows [Script Standard](https://docs.rsshub.app/joinus/advanced/script-standard) / 跟随 [路由规范](https://docs.rsshub.app/zh/joinus/advanced/script-standard)
- [ ] Documentation / 文档说明
- [x] Full text / 全文获取
- [ ] Use cache / 使用缓存
- [ ] Anti-bot or rate limit / 反爬/频率限制
- [ ] If yes, do your code reflect this sign? / 如果有, 是否有对应的措施?
- [x] [Date and time](https://docs.rsshub.app/joinus/advanced/pub-date) / [日期和时间](https://docs.rsshub.app/zh/joinus/advanced/pub-date)
- [x] Parsed / 可以解析
- [x] Correct time zone / 时区正确
- [ ] New package added / 添加了新的包
- [ ] `Puppeteer`

## Note / 说明

AhsokaTano26 committed:
feat(route):add CQU jwc notification
## Involved Issue / 该 PR 相关 Issue

Close #

## Example for the Proposed Route(s) / 路由地址示例
```routes
/cqu/jwc
```

## New RSS Route Checklist / 新 RSS 路由检查表

- [x] New Route / 新的路由
- [x] Follows [Script Standard](https://docs.rsshub.app/joinus/advanced/script-standard) / 跟随 [路由规范](https://docs.rsshub.app/zh/joinus/advanced/script-standard)
- [ ] Documentation / 文档说明
- [x] Full text / 全文获取
- [ ] Use cache / 使用缓存
- [ ] Anti-bot or rate limit / 反爬/频率限制
- [ ] If yes, do your code reflect this sign? / 如果有, 是否有对应的措施?
- [x] [Date and time](https://docs.rsshub.app/joinus/advanced/pub-date) / [日期和时间](https://docs.rsshub.app/zh/joinus/advanced/pub-date)
- [x] Parsed / 可以解析
- [x] Correct time zone / 时区正确
- [ ] New package added / 添加了新的包
- [ ] `Puppeteer`

## Note / 说明

TonyRL commented on lib/routes/cqu/jwc/jwc.ts:
```diff
@@ -0,0 +1,75 @@
+    radar: [
+        {
+        source: ['jwc.cqu.edu.cn/:path'],
+        target: '/cqu/jwc/:path',
```
```suggestion
        target: '/jwc/:path',
```

TonyRL commented on lib/routes/cqu/jwc/jwc.ts:
```diff
@@ -0,0 +1,75 @@
+        }
+    ],
+    name: '重庆大学本科教学信息网',
+    maintainers: ['TonyRL'],
```
This should fill in your GitHub ID, **NOT MINE**.

TonyRL commented on lib/routes/cqu/jwc/namespace.ts:
```diff
@@ -0,0 +1,7 @@
+
+export const namespace: Namespace = {
+    name: '重庆大学本科教学信息网',
+    url: 'https://jwc.cqu.edu.cn/',
```
```suggestion
    url: 'jwc.cqu.edu.cn',
```

TonyRL commented on lib/routes/cqu/jwc/jwc.ts:
```diff
@@ -0,0 +1,75 @@
+        target: '/cqu/jwc/:path',
+        }
+    ],
+    name: '重庆大学本科教学信息网',
```
Route name should not be the same as namespace name https://github.com/DIYgod/RSSHub/blob/fdeb57e2cb77b0cbdfb4a2626b6f51d22200b76e/lib/routes/cqu/jwc/namespace.ts#L4

AhsokaTano26 commented:
ok, thank you.I will change it soon.
&nbsp;


Ahsoka&nbsp;Tano
***@***.***

重庆大学
学生


&nbsp;




------------------&nbsp;原始邮件&nbsp;------------------
发件人: ***@***.***&gt;; 
发送时间: 2025年6月27日(星期五) 晚上9:28
收件人: ***@***.***&gt;; 
抄送: "Ahsoka ***@***.***&gt;; ***@***.***&gt;; 
主题: Re: [DIYgod/RSSHub] feat(route): add CQU jwc notification (PR #19468)














 
@TonyRL commented on this pull request.
 
 
In lib/routes/cqu/jwc/jwc.ts:
 &gt; +            description: '路径参数，默认为 `index/tzgg`', +            default: 'index/tzgg', +        }, +    }, +    features: { +        requireConfig: false, +        requirePuppeteer: false, +        antiCrawler: true, +        supportBT: false, +        supportPodcast: false, +        supportScihub: false, +    }, +    radar: [ +        { +        source: ['jwc.cqu.edu.cn/:path'], +        target: '/cqu/jwc/:path',  ⬇️ Suggested change -        target: '/cqu/jwc/:path', +        target: '/jwc/:path',  
 
In lib/routes/cqu/jwc/jwc.ts:
 &gt; +    features: { +        requireConfig: false, +        requirePuppeteer: false, +        antiCrawler: true, +        supportBT: false, +        supportPodcast: false, +        supportScihub: false, +    }, +    radar: [ +        { +        source: ['jwc.cqu.edu.cn/:path'], +        target: '/cqu/jwc/:path', +        } +    ], +    name: '重庆大学本科教学信息网', +    maintainers: ['TonyRL'],  
This should fill in your GitHub ID, NOT MINE.
 
 
In lib/routes/cqu/jwc/namespace.ts:
 &gt; @@ -0,0 +1,7 @@ +import type { Namespace } from '@/types'; + +export const namespace: Namespace = { +    name: '重庆大学本科教学信息网', +    url: 'https://jwc.cqu.edu.cn/',  ⬇️ Suggested change -    url: 'https://jwc.cqu.edu.cn/', +    url: 'jwc.cqu.edu.cn',  
 
In lib/routes/cqu/jwc/jwc.ts:
 &gt; +    }, +    features: { +        requireConfig: false, +        requirePuppeteer: false, +        antiCrawler: true, +        supportBT: false, +        supportPodcast: false, +        supportScihub: false, +    }, +    radar: [ +        { +        source: ['jwc.cqu.edu.cn/:path'], +        target: '/cqu/jwc/:path', +        } +    ], +    name: '重庆大学本科教学信息网',  
Route name should not be the same as namespace name https://github.com/DIYgod/RSSHub/blob/fdeb57e2cb77b0cbdfb4a2626b6f51d22200b76e/lib/routes/cqu/jwc/namespace.ts#L4
 
—
Reply to this email directly, view it on GitHub, or unsubscribe.
You are receiving this because you authored the thread.Message ID: ***@***.***&gt;

AhsokaTano26 committed:
feat(route):add CQU jwc notification
## Involved Issue / 该 PR 相关 Issue

Close #

## Example for the Proposed Route(s) / 路由地址示例
```routes
/cqu/jwc
```

## New RSS Route Checklist / 新 RSS 路由检查表

- [x] New Route / 新的路由
- [x] Follows [Script Standard](https://docs.rsshub.app/joinus/advanced/script-standard) / 跟随 [路由规范](https://docs.rsshub.app/zh/joinus/advanced/script-standard)
- [ ] Documentation / 文档说明
- [x] Full text / 全文获取
- [ ] Use cache / 使用缓存
- [ ] Anti-bot or rate limit / 反爬/频率限制
- [ ] If yes, do your code reflect this sign? / 如果有, 是否有对应的措施?
- [x] [Date and time](https://docs.rsshub.app/joinus/advanced/pub-date) / [日期和时间](https://docs.rsshub.app/zh/joinus/advanced/pub-date)
- [x] Parsed / 可以解析
- [x] Correct time zone / 时区正确
- [ ] New package added / 添加了新的包
- [ ] `Puppeteer`

## Note / 说明

TonyRL commented on lib/routes/cqu/jwc/namespace.ts:
```diff
@@ -0,0 +1,7 @@
+import type { Namespace } from '@/types';
+
+export const namespace: Namespace = {
+    name: '重庆大学本科教学信息网',
+    url: 'jwc.cqu.edu.cn',
```
```suggestion
    name: '重庆大学',
    url: 'cqu.edu.cn',
```
and the file path should be lib/routes/cqu/namespace.ts

TonyRL commented on lib/routes/cqu/jwc/jwc.ts:
```diff
@@ -0,0 +1,75 @@
+        target: '/jwc/:path',
+        }
+    ],
+    name: '重庆大学本科教学信息网通知',
```
Do not repeat namespace name 
```suggestion
    name: '本科教学信息网通知',
```
in route name

AhsokaTano26 committed:
feat(route):add CQU jwc notification
## Involved Issue / 该 PR 相关 Issue

Close #

## Example for the Proposed Route(s) / 路由地址示例
```routes
/cqu/jwc
```

## New RSS Route Checklist / 新 RSS 路由检查表

- [x] New Route / 新的路由
- [x] Follows [Script Standard](https://docs.rsshub.app/joinus/advanced/script-standard) / 跟随 [路由规范](https://docs.rsshub.app/zh/joinus/advanced/script-standard)
- [ ] Documentation / 文档说明
- [x] Full text / 全文获取
- [ ] Use cache / 使用缓存
- [ ] Anti-bot or rate limit / 反爬/频率限制
- [ ] If yes, do your code reflect this sign? / 如果有, 是否有对应的措施?
- [x] [Date and time](https://docs.rsshub.app/joinus/advanced/pub-date) / [日期和时间](https://docs.rsshub.app/zh/joinus/advanced/pub-date)
- [x] Parsed / 可以解析
- [x] Correct time zone / 时区正确
- [ ] New package added / 添加了新的包
- [ ] `Puppeteer`

## Note / 说明

TonyRL reviewed (commented):
The file path **should be** lib/routes/cqu/namespace.ts

https://github.com/DIYgod/RSSHub/pull/19468#discussion_r2349700304

AhsokaTano26 committed:
feat(route):add CQU jwc notification
## Involved Issue / 该 PR 相关 Issue

Close #

## Example for the Proposed Route(s) / 路由地址示例
```routes
/cqu/jwc
```

## New RSS Route Checklist / 新 RSS 路由检查表

- [x] New Route / 新的路由
- [x] Follows [Script Standard](https://docs.rsshub.app/joinus/advanced/script-standard) / 跟随 [路由规范](https://docs.rsshub.app/zh/joinus/advanced/script-standard)
- [ ] Documentation / 文档说明
- [x] Full text / 全文获取
- [ ] Use cache / 使用缓存
- [ ] Anti-bot or rate limit / 反爬/频率限制
- [ ] If yes, do your code reflect this sign? / 如果有, 是否有对应的措施?
- [x] [Date and time](https://docs.rsshub.app/joinus/advanced/pub-date) / [日期和时间](https://docs.rsshub.app/zh/joinus/advanced/pub-date)
- [x] Parsed / 可以解析
- [x] Correct time zone / 时区正确
- [ ] New package added / 添加了新的包
- [ ] `Puppeteer`

## Note / 说明

AhsokaTano26 committed:
feat(route):add CQU jwc notification
## Involved Issue / 该 PR 相关 Issue

Close #

## Example for the Proposed Route(s) / 路由地址示例
```routes
/cqu/jwc
```

## New RSS Route Checklist / 新 RSS 路由检查表

- [x] New Route / 新的路由
- [x] Follows [Script Standard](https://docs.rsshub.app/joinus/advanced/script-standard) / 跟随 [路由规范](https://docs.rsshub.app/zh/joinus/advanced/script-standard)
- [ ] Documentation / 文档说明
- [x] Full text / 全文获取
- [ ] Use cache / 使用缓存
- [ ] Anti-bot or rate limit / 反爬/频率限制
- [ ] If yes, do your code reflect this sign? / 如果有, 是否有对应的措施?
- [x] [Date and time](https://docs.rsshub.app/joinus/advanced/pub-date) / [日期和时间](https://docs.rsshub.app/zh/joinus/advanced/pub-date)
- [x] Parsed / 可以解析
- [x] Correct time zone / 时区正确
- [ ] New package added / 添加了新的包
- [ ] `Puppeteer`

## Note / 说明
